x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/details/index.tsx

```diff
@@ -450,7 +479,20 @@ const RuleDetailsPageComponent: React.FC<DetectionEngineComponentProps> = ({
     ),
     [isExistingRule, ruleDetailTab, setRuleDetailTab, pageTabs]
   );
-  const ruleIndices = useMemo(() => rule?.index ?? DEFAULT_INDEX_PATTERN, [rule?.index]);
+
+  const [ruleIndices, setRuleIndices] = useState<string[]>(rule?.index ?? DEFAULT_INDEX_PATTERN);
+
+  // TODO: Revisit this logic. I think we can remove the await data
+  useEffect(() => {
+    const fetchDataViews = async () => {
+      if (rule?.data_view_id != null && rule?.data_view_id !== '') {
+        const dataView = await data.dataViews.get(rule?.data_view_id);
+        setRuleIndices(dataView.title.split(','));
+      }
+    };
+
+    fetchDataViews();
+  }, [rule?.index, rule?.data_view_id, data]);
```

Comment:
Does this need to happen here, or can it be pushed further down into `StepDefineRule`? Maybe even all the way to the read-only viewer component?

Reply:
This is a good idea. I will play around with this today. I think this could have resolved a bug I ran into last week and explain why some of the tests were failing in CI.